src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilder.cs

```diff
@@ -251,8 +252,7 @@ private ServerFactory CreateServerFactory()
 
         private IServerMonitorFactory CreateServerMonitorFactory()
         {
-            var serverMonitorConnectionSettings = _connectionSettings
-                .With(authenticators: new IAuthenticator[] { });
+            var serverMonitorConnectionSettings = _connectionSettings.With(authenticatorsConfigurator: Optional.Create<Func<IEnumerable<IAuthenticator>>>(() => new IAuthenticator[0]));
```

Comment:
the main idea of the PR change is that now we don't save authenticators itself, instead, we save a delegate which can create authenticators by demand